importing necessary libraries and packages like pandas and numpy etc

In [1]:
import pandas as pd
import numpy as np

reading the data into a dataframe

In [2]:
oj_data=pd.read_csv("oj.csv",encoding="latin")

viewing first five rows of dataframe

In [3]:
oj_data.head()

store      brand  week   logmove  feat  price     AGE60      EDUC   ETHNIC  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      INCOME   HHLARGE   WORKWOM   HVAL150  SSTRDIST   SSTRVOL  CPDIST5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    CPWVOL5  
0  0.376927  
1  0.376927  
2  0.376927  
3  0.376927  
4  0.376927

1.Find the dimensions of the oj data set

In [4]:
oj_data.shape

(28947, 17)

***the oj_data has 28947 rows and 17 columns***

2.	Find the structure of the data set

In [5]:
oj_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28947 entries, 0 to 28946
Data columns (total 17 columns):
store       28947 non-null int64
brand       28947 non-null object
week        28947 non-null int64
logmove     28947 non-null float64
feat        28947 non-null int64
price       28947 non-null float64
AGE60       28947 non-null float64
EDUC        28947 non-null float64
ETHNIC      28947 non-null float64
INCOME      28947 non-null float64
HHLARGE     28947 non-null float64
WORKWOM     28947 non-null float64
HVAL150     28947 non-null float64
SSTRDIST    28947 non-null float64
SSTRVOL     28947 non-null float64
CPDIST5     28947 non-null float64
CPWVOL5     28947 non-null float64
dtypes: float64(13), int64(3), object(1)
memory usage: 3.8+ MB


*** info tells about details of each column like the total number of entries,datatype etc***

3.Find out the column names in the data set

In [6]:
list(oj_data.columns)

['store',
 'brand',
 'week',
 'logmove',
 'feat',
 'price',
 'AGE60',
 'EDUC',
 'ETHNIC',
 'INCOME',
 'HHLARGE',
 'WORKWOM',
 'HVAL150',
 'SSTRDIST',
 'SSTRVOL',
 'CPDIST5',
 'CPWVOL5']

***we use the keyword column to get the column names and preview it in form of list using list()***

4.Describe the data set

In [7]:
oj_data.describe()

store          week       logmove          feat         price  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean      80.883511    100.459944      9.167864      0.237261      2.282488   
std       35.576511     34.692314      1.019378      0.425411      0.648001   
min        2.000000     40.000000      4.158883      0.000000      0.520000   
25%       53.000000     70.000000      8.489616      0.000000      1.790000   
50%       86.000000    101.000000      9.034080      0.000000      2.170000   
75%      111.000000    130.000000      9.764685      0.000000      2.730000   
max      137.000000    160.000000     13.482016      1.000000      3.870000   

              AGE60          EDUC        ETHNIC        INCOME       HHLARGE  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean       0.173130      0.225220      0.155557     10.616735      0.115602   
std        0.061872      0.109945      0.187581      0.282314      0.030168   
min        0.058054      0.049550      0.024247      9.867083      0.013506   
25%        0.122100      0.145985      0.041910     10.456079      0.097938   
50%        0.170655      0.229390      0.074656     10.635326      0.111221   
75%        0.213949      0.284395      0.187761     10.796960      0.135168   
max        0.307398      0.528362      0.995691     11.236197      0.216354   

            WORKWOM       HVAL150      SSTRDIST       SSTRVOL       CPDIST5  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean       0.359178      0.343766      5.097274      1.207317      2.120359   
std        0.052673      0.239028      3.472386      0.526528      0.729828   
min        0.244463      0.002509      0.132097      0.400000      0.772530   
25%        0.312636      0.123486      2.767046      0.727273      1.626192   
50%        0.355635      0.346154      4.650687      1.115385      1.963412   
75%        0.402313      0.528313      6.650602      1.538462      2.533672   
max        0.472308      0.916699     17.855951      2.571429      4.107902   

            CPWVOL5  
count  28947.000000  
mean       0.438914  
std        0.219248  
min        0.094562  
25%        0.271673  
50%        0.383227  
75%        0.560240  
max        1.143367

describe() gives the summary statistics of each of the numerical columns in the data
it displays count,mean,standard deviation,max,min quantilese etc

5. Update the column names of the data set, such that every column name is in lowercase

In [8]:
oj_data.columns=[col.lower() for col in list(oj_data.columns)]
list(oj_data.columns)

['store',
 'brand',
 'week',
 'logmove',
 'feat',
 'price',
 'age60',
 'educ',
 'ethnic',
 'income',
 'hhlarge',
 'workwom',
 'hval150',
 'sstrdist',
 'sstrvol',
 'cpdist5',
 'cpwvol5']

***to update column names into lower case we use lower() and display them in list***

1.Fetch the first row 3rd column from the data set

In [9]:
oj_data.iloc[0,2]

40

***iloc is used to selecting data by row numbers***

2.Fetch the first, second and Third columns of the oj data frame

In [10]:
oj_data.iloc[:,[0,1,2]].head()

store      brand  week
0      2  tropicana    40
1      2  tropicana    46
2      2  tropicana    47
3      2  tropicana    48
4      2  tropicana    50

the first second and third columns are store,brand,week
we use iloc to retrive data 

3.Fetch the first, second, eighth and the 456th rows of the 1st, third and the sixth columns of the data frame

In [11]:
oj_data.iloc[[0,1,7,455],[0,2,5]] 

store  week  price
0        2    40   3.87
1        2    46   3.87
7        2    53   3.29
455      5    51   2.99

first second and sixth columns are store week and price respectively
first,second,eighth and 456th rows are retrived using iloc by specifying the same as row and column numbers

4.Fetch the first 5 rows of the brand column of the dataset

In [12]:
oj_data['brand'].head()

0    tropicana
1    tropicana
2    tropicana
3    tropicana
4    tropicana
Name: brand, dtype: object

***the top 5 rows of brand column are selected***

5.Fetch top 5 rows of the brand, week and feat details 

In [13]:
oj_data[['brand','week','feat']].head()

brand  week  feat
0  tropicana    40     0
1  tropicana    46     0
2  tropicana    47     0
3  tropicana    48     0
4  tropicana    50     0

***top 5 rows of brand,week and feat details***

6.Fetch the details of all the distinct stores

In [14]:
oj_data["store"].unique()

array([  2,   5,   8,   9,  12,  14,  18,  21,  28,  32,  33,  40,  44,
        45,  47,  48,  49,  50,  51,  52,  53,  54,  56,  59,  62,  64,
        67,  68,  70,  71,  72,  73,  74,  75,  76,  77,  78,  80,  81,
        83,  84,  86,  88,  89,  90,  91,  92,  93,  94,  95,  97,  98,
       100, 101, 102, 103, 104, 105, 106, 107, 109, 110, 111, 112, 113,
       114, 115, 116, 117, 118, 119, 121, 122, 123, 124, 126, 128, 129,
       130, 131, 132, 134, 137], dtype=int64)

***using "store"column unique() will retrive details of all the distinct stores ***

7.Fetch all the observations for Tropicana brand

In [15]:
oj_data[oj_data['brand']=='tropicana'].head()

store      brand  week   logmove  feat  price     age60      educ   ethnic  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      income   hhlarge   workwom   hval150  sstrdist   sstrvol  cpdist5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    cpwvol5  
0  0.376927  
1  0.376927  
2  0.376927  
3  0.376927  
4  0.376927

***all the details of tropicana brand***

8.Fetch all the observations for Tropicana brand using query function

In [16]:
oj_data.query("brand=='tropicana'").head()

store      brand  week   logmove  feat  price     age60      educ   ethnic  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      income   hhlarge   workwom   hval150  sstrdist   sstrvol  cpdist5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    cpwvol5  
0  0.376927  
1  0.376927  
2  0.376927  
3  0.376927  
4  0.376927

*** retriving all details of tropicana brand using query function***

9.Fetch store,brand and price observations for the brand toppicana

In [17]:
oj_data[oj_data.brand=='tropicana'][['store','brand','price']].head()

store      brand  price
0      2  tropicana   3.87
1      2  tropicana   3.87
2      2  tropicana   3.87
3      2  tropicana   3.87
4      2  tropicana   3.87

***for tropicana brand the store and price details are selected***

10.Fetch last 5  observations for those who have bought Tropicana or dominics

In [18]:
oj_data[(oj_data.brand == 'tropicana') | (oj_data.brand == 'dominicks')].tail(5)

store      brand  week    logmove  feat  price     age60      educ  \
28942    137  dominicks   156   7.987524     0   2.00  0.209602  0.528362   
28943    137  dominicks   157   8.502689     0   1.94  0.209602  0.528362   
28944    137  dominicks   158   9.505991     0   1.59  0.209602  0.528362   
28945    137  dominicks   159  10.927376     0   1.49  0.209602  0.528362   
28946    137  dominicks   160   8.859363     0   1.75  0.209602  0.528362   

        ethnic    income   hhlarge   workwom   hval150  sstrdist   sstrvol  \
28942  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28943  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28944  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28945  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28946  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   

       cpdist5   cpwvol5  
28942  0.77253  0.333761  
28943  0.77253  0.333761  
28944  0.77253  0.333761  
28945  0.77253  0.333761  
28946  0.77253  0.333761

***details of last 5 observations who have bought tropicana or dominics***

11.Fetch the first 5 observations of income, brand, price observations with Tropicana brand 
without feature advertisement

In [19]:
oj_data.query("(brand=='tropicana')&(feat==0)")[["income","brand","price"]].head()

income      brand  price
0  10.553205  tropicana   3.87
1  10.553205  tropicana   3.87
2  10.553205  tropicana   3.87
3  10.553205  tropicana   3.87
4  10.553205  tropicana   3.87

*** using query function the details of 5 observations of income,brand,price with tropicana brand without feature advertisement***

12.Add a new column to the dataframe normInc which is the normalised value of income

In [20]:
oj_data['normInc'] = ((oj_data['income'])-oj_data['income'].mean())/(oj_data['income'].max()-oj_data['income'].min())

***inorder to find the normalised income the formula used is
Normalised income= (income-mean(income))/(max(income)-min(income))
the caluculated normalised income is stored as new column in the dataframe with the name normInc***
we can check the addition of new column by using .column and its values in the data set using head() or viewing shape of data farme

In [21]:
oj_data.columns

Index(['store', 'brand', 'week', 'logmove', 'feat', 'price', 'age60', 'educ',
       'ethnic', 'income', 'hhlarge', 'workwom', 'hval150', 'sstrdist',
       'sstrvol', 'cpdist5', 'cpwvol5', 'normInc'],
      dtype='object')

In [22]:
oj_data.head()

store      brand  week   logmove  feat  price     age60      educ   ethnic  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      income   hhlarge   workwom   hval150  sstrdist   sstrvol  cpdist5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    cpwvol5   normInc  
0  0.376927 -0.046402  
1  0.376927 -0.046402  
2  0.376927 -0.046402  
3  0.376927 -0.046402  
4  0.376927 -0.046402

In [23]:
oj_data.shape

(28947, 18)

14.Fetch the observations for the tropicana brand excluding the store, price and income fields

In [24]:
oj_data.query("brand == 'tropicana'").drop(["price","store","income"],1).head()

brand  week   logmove  feat     age60      educ   ethnic   hhlarge  \
0  tropicana    40  9.018695     0  0.232865  0.248935  0.11428  0.103953   
1  tropicana    46  8.723231     0  0.232865  0.248935  0.11428  0.103953   
2  tropicana    47  8.253228     0  0.232865  0.248935  0.11428  0.103953   
3  tropicana    48  8.987197     0  0.232865  0.248935  0.11428  0.103953   
4  tropicana    50  9.093357     0  0.232865  0.248935  0.11428  0.103953   

    workwom   hval150  sstrdist   sstrvol  cpdist5   cpwvol5   normInc  
0  0.303585  0.463887  2.110122  1.142857  1.92728  0.376927 -0.046402  
1  0.303585  0.463887  2.110122  1.142857  1.92728  0.376927 -0.046402  
2  0.303585  0.463887  2.110122  1.142857  1.92728  0.376927 -0.046402  
3  0.303585  0.463887  2.110122  1.142857  1.92728  0.376927 -0.046402  
4  0.303585  0.463887  2.110122  1.142857  1.92728  0.376927 -0.046402

***using query() and drop() functions, details for tropicana brand excluding price store and income fields are retrived***

15.Sort the Data in the increasing order of the week

In [25]:
oj_data.sort_values(by="week",ascending=True).head()

store        brand  week   logmove  feat  price     age60      educ  \
0          2    tropicana    40  9.018695     0   3.87  0.232865  0.248935   
8141      59  minute.maid    40  7.655391     0   2.62  0.110819  0.233036   
8025      59    tropicana    40  8.489616     0   3.19  0.110819  0.233036   
26035    124  minute.maid    40  8.269757     0   3.17  0.119626  0.261876   
7907      56    dominicks    40  8.565602     1   1.59  0.192889  0.237551   

         ethnic     income   hhlarge   workwom   hval150  sstrdist   sstrvol  \
0      0.114280  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857   
8141   0.024247  10.715040  0.140676  0.390696  0.292652  0.217275  1.000000   
8025   0.024247  10.715040  0.140676  0.390696  0.292652  0.217275  1.000000   
26035  0.572356  10.258957  0.124950  0.348519  0.416316  6.328747  0.727273   
7907   0.041356  10.831825  0.105928  0.362168  0.578125  4.865721  0.533333   

        cpdist5   cpwvol5   normInc  
0      1.927280  0.376927 -0.046402  
8141   3.331154  0.395539  0.071802  
8025   3.331154  0.395539  0.071802  
26035  1.793796  0.202169 -0.261321  
7907   2.998578  0.496127  0.157102

***by using the sort_values() function we sort the data in ascending order of week***

16. Sort the data in the decreasing order of Income and increasing order of week

In [26]:
oj_data.sort_values(by=["income","week"],ascending=[False,True]).head()

store        brand  week    logmove  feat  price     age60     educ  \
8373     62    tropicana    40   9.373819     0   3.87  0.222534  0.51776   
8490     62  minute.maid    40   8.449343     0   3.17  0.222534  0.51776   
8607     62    dominicks    40   8.050703     1   1.59  0.222534  0.51776   
8374     62    tropicana    41   9.368369     0   3.87  0.222534  0.51776   
8491     62  minute.maid    41  10.667652     1   1.99  0.222534  0.51776   

        ethnic     income   hhlarge   workwom   hval150  sstrdist   sstrvol  \
8373  0.026511  11.236197  0.103979  0.322765  0.916699  5.452685  0.705882   
8490  0.026511  11.236197  0.103979  0.322765  0.916699  5.452685  0.705882   
8607  0.026511  11.236197  0.103979  0.322765  0.916699  5.452685  0.705882   
8374  0.026511  11.236197  0.103979  0.322765  0.916699  5.452685  0.705882   
8491  0.026511  11.236197  0.103979  0.322765  0.916699  5.452685  0.705882   

      cpdist5   cpwvol5   normInc  
8373  2.18405  0.201722  0.452455  
8490  2.18405  0.201722  0.452455  
8607  2.18405  0.201722  0.452455  
8374  2.18405  0.201722  0.452455  
8491  2.18405  0.201722  0.452455

***using sort_values() sorting the data in decreasing order of income and increasing order of week***

17.Find the mean of the Juice price for each brand

In [27]:
oj_data[["brand","price"]].groupby("brand").mean()

price
brand                
dominicks    1.735809
minute.maid  2.241162
tropicana    2.870493

***finding the mean of juice price for each brand using groupby()***

-->>***can also be done using aggregate function***

In [28]:
oj_data.groupby("brand").agg({"price":"mean"})

price
brand                
dominicks    1.735809
minute.maid  2.241162
tropicana    2.870493

18.Find the mean, sum, std of price for every brand

In [29]:
oj_data[["brand","price"]].groupby("brand").agg(["mean","sum","std"])

price                    
                 mean       sum       std
brand                                    
dominicks    1.735809  16748.82  0.385838
minute.maid  2.241162  21624.97  0.404515
tropicana    2.870493  27697.39  0.548558

***taking price and brand alone and finding the mean,sum and standard deviation price of each brand using groupby() and agg() functions***

19.For every brand find the mean price and sum of income

In [30]:
oj_data.groupby("brand").agg({"price":"mean","income":"sum"})

price         income
brand                               
dominicks    1.735809  102440.872522
minute.maid  2.241162  102440.872522
tropicana    2.870493  102440.872522

***the mean price and sum of income for every brand are caluculated using groupby() and agg() function***

20.Find the average income for each brand and at each store

In [31]:
oj_data.groupby(["brand","store"]).agg({"income":"mean"}).head()

income
brand     store           
dominicks 2      10.553205
          5      10.922371
          8      10.597010
          9      10.787152
          12      9.996659

***for each brand at each store the average income is caluculated using groupby() and agg() functions***

21a.Mean and std deviation of the income for each brand

In [32]:
oj_data[["brand","income"]].groupby("brand").agg(["mean","std","count"])

income                
                  mean       std count
brand                                 
dominicks    10.616735  0.282323  9649
minute.maid  10.616735  0.282323  9649
tropicana    10.616735  0.282323  9649

***mean and standard deviation of income for each brand is caluculated using groupby() and agg()***

21b.mean of price for income >=10.5

In [33]:
oj_data.query("income>=10.5")[["price"]].mean()

price    2.270229
dtype: float64

***for income>=10.5 the mean price is caluculated***

21c.mean of price for income >=10.5 for each brand

In [34]:
oj_data.groupby(["brand","feat"]).agg({"store":"size"})

store
brand       feat       
dominicks   0      7169
            1      2480
minute.maid 0      6865
            1      2784
tropicana   0      8045
            1      1604

***mean price for income>=10.5 for each brand is caluculated***

22.Find the Cross tabulation of brands and feature advertisement

In [35]:
dat=oj_data[["brand","feat","store"]]
dat.groupby(["brand","feat"]).count()

store
brand       feat       
dominicks   0      7169
            1      2480
minute.maid 0      6865
            1      2784
tropicana   0      8045
            1      1604

***the total observations for each brand and for every feature advertisement category***

In [36]:
pd.crosstab(oj_data['brand'],oj_data['feat'])

feat            0     1
brand                  
dominicks    7169  2480
minute.maid  6865  2784
tropicana    8045  1604

***crosstabulation of brands and feature advertisement***